core/src/main/java/org/jdbi/v3/core/argument/BuiltInArgumentFactory.java

```diff
@@ -86,6 +88,8 @@
         register(map, double.class, Types.DOUBLE, PreparedStatement::setDouble);
         register(map, Float.class, Types.FLOAT, PreparedStatement::setFloat);
         register(map, float.class, Types.FLOAT, PreparedStatement::setFloat);
+        register(map, Inet4Address.class, Types.OTHER, (p, i, v) -> p.setObject(i, v.getHostAddress(), Types.OTHER));
+        register(map, Inet6Address.class, Types.OTHER, (p, i, v) -> p.setObject(i, v.getHostAddress(), Types.OTHER));
```

Comment:
The primitive factory doesn't consider type hierarchy, only a lookup of the exact class passed in.  `InetAddress` is an abstract class so it would never match.  We could improve this logic, but it's nonsense for most types (`int` doesn't have a superclass) and it'd be more verbose than just enumerating the implementation types here for now.